admin/load.php

```diff
@@ -609,18 +641,97 @@ function perflab_deactivate_plugin() {
 }
 add_action( 'admin_action_perflab_deactivate_plugin', 'perflab_deactivate_plugin' );
 
+// WordPress AJAX action to handle the button click event.
+add_action( 'wp_ajax_perflab_install_activate_standalone_plugins', 'perflab_install_activate_standalone_plugins_callback' );
+
 /**
- * Callback function hooked to admin_notices to render plugin activate/deactivate notices.
+ * Handles the standalone plugin install and activation via AJAX.
  *
  * @since n.e.x.t
- *
- * @return void
  */
-function perflab_plugin_admin_notices() {
-	if ( 'settings_page_perflab-modules' !== get_current_screen()->id ) {
-		return;
+function perflab_install_activate_standalone_plugins_callback() {
+	if ( ! wp_verify_nonce( $_REQUEST['nonce'], 'perflab-install-activate-plugins' ) ) {
+		$status['errorMessage'] = esc_html__( 'Invalid nonce: Please refresh and try again.', 'performance-lab' );
+		wp_send_json_error( $status );
+	}
+
+	if ( ! current_user_can( 'install_plugins' ) || ! current_user_can( 'activate_plugins' ) ) {
+		$status['errorMessage'] = esc_html__( 'Sorry, you are not allowed to manage plugins for this site. Please contact the administrator.', 'performance-lab' );
+		wp_send_json_error( $status );
+	}
+
+	if ( ! function_exists( 'plugins_api' ) ) {
+		require_once ABSPATH . 'wp-admin/includes/plugin-install.php';
+	}
+	require_once ABSPATH . 'wp-admin/includes/class-wp-upgrader.php';
+	require_once ABSPATH . 'wp-admin/includes/class-wp-ajax-upgrader-skin.php';
+
+	$plugins_to_activate = perflab_get_active_modules_with_standalone_plugins();
+	$modules             = perflab_get_module_settings();
+	$plugins             = get_plugins();
+	$status              = array();
+
+	foreach ( $plugins_to_activate as $module_slug ) {
+
+		// Skip checking for already activated plugin.
+		if ( perflab_is_standalone_plugin_loaded( $module_slug ) ) {
+			continue;
+		}
+
+		$plugin_slug     = basename( $module_slug );
+		$plugin_basename = $plugin_slug . '/load.php';
+		$api             = perflab_query_plugin_info( $plugin_slug );
+
+		// Return early if plugin API return an error.
+		if ( ! $api ) {
+			$status['errorMessage'] = __( 'An unexpected error occurred. Something may be wrong with WordPress.org or this server&#8217;s configuration.', 'performance-lab' );
```

Comment:
Would it not be better to not use an HTML entity in this case? Then you wouldn't have to decode it later with `perflab_replace_html_entity()` in JS. But you or @swissspidy would know better in regards to i18n best practices. 

```suggestion
			$status['errorMessage'] = __( 'An unexpected error occurred. Something may be wrong with WordPress.org or this server’s configuration.', 'performance-lab' );
```

Reply:
Otherwise, the decoding of the HTML entities could be done right here in PHP:

```suggestion
			$status['errorMessage'] = html_entity_decode( __( 'An unexpected error occurred. Something may be wrong with WordPress.org or this server&#8217;s configuration.', 'performance-lab' ), ENT_QUOTES );
```

Reply:
Well I‘d normally recommend using the wp.i18n functions in JS instead of passing strings from PHP to JS. Then you don‘t really have to deal with that.

Reply:
Thanks both for the feedback. As we have different  messages for error in JS do we need to add them conditionally then use `wp.i18n.__( 'Error message', 'performance-lab' )` function to print them in JS?

Reply:
I see, you're sending them to JS with `wp_send_json_*`. That should be fine to keep as-is 👍

Reply:
Just to know do we need to address this and https://github.com/WordPress/performance/pull/899#discussion_r1428388934 feedbacks? Thanks!

Reply:
Yes, Weston's feedback. Just my comment above can be ignored :)

I suggest using the `html_entity_decode` approach suggested above.

Reply:
Thanks. Addressed in https://github.com/WordPress/performance/pull/899/commits/3e0256c45bdcc02fc05e5563b8d4d4f6c571fa9a